community/lucene-index/src/test/java/org/neo4j/kernel/impl/api/constraints/ConstraintCreationIT.java

```diff
@@ -0,0 +1,105 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.api.constraints;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.neo4j.graphdb.ConstraintViolationException;
+import org.neo4j.graphdb.DependencyResolver;
+import org.neo4j.graphdb.DynamicLabel;
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.graphdb.factory.GraphDatabaseFactory;
+import org.neo4j.helpers.collection.Iterables;
+import org.neo4j.kernel.GraphDatabaseAPI;
+import org.neo4j.kernel.api.index.util.FailureStorage;
+import org.neo4j.kernel.impl.api.index.IndexingService;
+import org.neo4j.kernel.impl.nioneo.xa.NeoStoreProvider;
+import org.neo4j.kernel.monitoring.Monitors;
+import org.neo4j.test.EphemeralFileSystemRule;
+import org.neo4j.test.TargetDirectory;
+import org.neo4j.test.TestGraphDatabaseFactory;
+import org.neo4j.test.impl.EphemeralFileSystemAbstraction;
+import org.neo4j.tooling.GlobalGraphOperations;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class ConstraintCreationIT
+{
+    private static final Label LABEL = DynamicLabel.label( "label1" );
+    private GraphDatabaseAPI db;
+
+    @Rule
+    public final TargetDirectory.TestDirectory testDirectory = TargetDirectory.testDirForTest( getClass() );
+
+    @Test
+    public void shouldNotLeaveLuceneIndexFilesHangingAroundIfConstraintCreationFails()
+    {
+        // given
+        GraphDatabaseAPI db =
+                (GraphDatabaseAPI) new GraphDatabaseFactory().newEmbeddedDatabase( testDirectory.directory().getPath() );
+
+        System.out.println(db.getStoreDir());
+
+        try ( Transaction tx = db.beginTx() )
+        {
+            for ( int i = 0; i < 2; i++ )
+            {
+                Node node1 = db.createNode( LABEL );
+                node1.setProperty( "prop", true );
+            }
+
+            tx.success();
+        }
+
+        // when
+        try ( Transaction tx = db.beginTx() )
+        {
+            db.schema().constraintFor( LABEL ).assertPropertyIsUnique( "prop" ).create();
+            fail("Should have failed with ConstraintViolationException");
+            tx.success();
+        }
+        catch ( ConstraintViolationException ignored )  { }
+
+        // then
+        try(Transaction tx = db.beginTx())
+        {
+            assertEquals(0, Iterables.count(db.schema().getIndexes() ));
+        }
+
+        File[] files = new File(db.getStoreDir(), "schema/index/lucene/1/").listFiles();
```

Comment:
Can we get this "schema/index/lucene/1/" from somewhere instead hard coding. If we changed the dir layout sometime in the future, this test could still be false green.